extensions/cornerstone/src/components/ModalityLoadBadge/ModalityLoadBadge.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { ViewportActionButton } from '@ohif/ui-next';
+import { Icons, Tooltip, TooltipTrigger, TooltipContent } from '@ohif/ui-next';
+import { useSystem } from '@ohif/core/src';
+import { useViewportDisplaySets } from '../../hooks/useViewportDisplaySets';
+import { useMeasurementTracking } from '../../hooks/useMeasurementTracking';
+
+/**
+ * ModalityLoadBadge displays the status and actionable buttons for viewports containing
+ * special displaySets (SR, SEG, RTSTRUCT) or when tracking measurements
+ */
+function ModalityLoadBadge({ viewportId }: { viewportId: string }) {
+  const { commandsManager } = useSystem();
+  const { t } = useTranslation('Common');
+  const loadStr = t('LOAD');
+
+  const { isTracked, isLocked } = useMeasurementTracking({ viewportId });
+
+  const { backgroundDisplaySet, overlayDisplaySets } = useViewportDisplaySets(viewportId);
+
+  const [specialDisplaySet, setSpecialDisplaySet] = useState(null);
+
+  const allDisplaySets = useMemo(() => {
+    return [backgroundDisplaySet, ...overlayDisplaySets].filter(Boolean);
+  }, [backgroundDisplaySet, overlayDisplaySets]);
+
+  useEffect(() => {
+    const displaySet = allDisplaySets.find(
+      ds => ds?.Modality === 'SEG' || ds?.Modality === 'RTSTRUCT' || ds?.Modality === 'SR'
```

Comment:
Can you use the overlay key here instead, or another attribute?  Maybe isRehydrateable would work?